sentry_sdk/hub.py

```diff
@@ -441,38 +441,138 @@ def start_span(
+                "Deprecated: use start_transaction to start transactions and "
+                "Transaction.start_child to start spans."
+            )
+            if isinstance(span, Transaction):
+                logger.warning(deprecation_msg)
+                return self.start_transaction(span)
+            if "transaction" in kwargs:
+                logger.warning(deprecation_msg)
+                name = kwargs.pop("transaction")
+                return self.start_transaction(name=name, **kwargs)
+
+        # XXX: this is not very useful because
+        #
+        #     with hub.start_span(Span(...)):
+        #         pass
+        #
+        # is equivalent to the simpler
+        #
+        #     with Span(...):
+        #         pass
+        if span is not None:
+            return span
 
         kwargs.setdefault("hub", self)
 
-        if span is None:
-            span = scope.span
-            if span is not None:
-                span = span.new_span(**kwargs)
-            else:
-                span = Span(**kwargs)
+        span = self.scope.span
+        if span is not None:
+            return span.start_child(**kwargs)
+
+        # XXX: returning a detached span here means whatever span tree built
+        # from it will be eventually discarded.
+        #
+        # This is also not very useful because
+        #
+        #     with hub.start_span(op="..."):
+        #         pass
+        #
+        # is equivalent, when there is no span in the scope, to the simpler
+        #
+        #     with Span(op="..."):
+        #         pass
+        return Span(**kwargs)
+
+    def start_transaction(
+        self,
+        transaction=None,  # type: Optional[Transaction]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> Transaction
+        """
+        Start and return a transaction.
+
+        Start an existing transaction if given, otherwise create and start a new
+        transaction with kwargs.
+
+        This is the entry point to manual tracing instrumentation.
+
+        A tree structure can be built by adding child spans to the transaction,
+        and child spans to other spans. To start a new child span within the
+        transaction or any span, call the respective `.start_child()` method.
+
+        Every child span must be finished before the transaction is finished,
+        otherwise the unfinished spans are discarded.
+
+        When used as context managers, spans and transactions are automatically
+        finished at the end of the `with` block. If not using context managers,
+        call the `.finish()` method.
+
+        When the transaction is finished, it will be sent to Sentry with all its
+        finished child spans.
+        """
+        # XXX: should we always set transaction.hub = self?
+        # In a multi-hub program, what does it mean to write
+        #     hub1.start_transaction(Transaction(hub=hub2))
+        # ? Should the transaction be on hub1 or hub2?
```

Comment:
Hmm, this is tricky. I think if we keep our current implementation, the transaction should be on `hub1` and we should not allow setting hub through `kwargs`. If we decide the change the API, we can revisit this.